src/main.rs

```diff
@@ -67,7 +68,16 @@ pub fn main() {
         return;
     }
 
-    if let Err(code) = process(env::args().skip(2)) {
+    // if we run "cargo clippy" (as cargo subcommand), we have to strip "cargo clippy" (first 2 args)
+    // but if we are run via "..../target/debug/cargo-clippy", only ommit the first arg, the second one
+    // might be a normal cmdline arg already (which we don't want to ommit)
+    let args = if let Some(first_arg) = env::args().next() && (first_arg.ends_with("cargo-clippy") || first_arg.ends_with("cargo-clippy.exe")) {
```

Comment:
We can avoid any breaking change by checking if the 2nd arg is `clippy`, rather than the first being an invocation of `cargo-clippy[.exe]`

Reply:
ah so that `cargo clippy` but also `cargo-clippy clippy` still works? clever! :D

I'll revisit this tomorrow after a good portion of sleep :)

Reply:
That doesn't avoid the breaking change. Anything that uses `cargo-clippy --` would break. Or anything else for that matter.

Reply:
That is what is happening for `lintcheck` for example

Reply:
Ah right since any arg can currently be used as the separator

Reply:
`cargo clippy` does execute with the first arg as `.../cargo-clippy` though, so we'd still need a different approach

```
execve("/home/alex/.cargo/bin/cargo", ["cargo", "clippy"], 0xffffd242db68 /* 49 vars */) = 0
execve("/home/alex/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo", ["/home/alex/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo", "clippy"], 0xaaaae39b8940 /* 54 vars */) = 0
execve("/home/alex/.cargo/bin/cargo-clippy", ["/home/alex/.cargo/bin/cargo-clippy", "clippy"], 0xaaab0d612f40 /* 57 vars */) = 0
execve("/home/alex/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo-clippy", ["/home/alex/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo-clippy", "clippy"], 0xaaaafea6a490 /* 57 vars */) = 0```

Reply:
I just checked clippy `1.50` which was the lastest version when I wrote lintcheck 2 years ago and it looks like 
````
RUSTC_BOOTSTRAP=1 ~/.rustup/toolchains/1.50-x86_64-unknown-linux-gnu/bin/cargo-clippy --fix -Zunstable-options
````
actually never worked as intended :no_mouth:  :sweat_smile: :sweat_smile: